docs/editors/vim.md

```diff
@@ -43,47 +54,143 @@ following settings.
 exist yet.
 
-```scala mdoc:bootstrap:metals-vim vim-lsc
+```scala mdoc:bootstrap:metals-vim coc
 
 ```
 
-The `-Dmetals.client=vim-lsc` flag is important since it configures Metals for
-usage with the `vim-lsc` client.
+The `-Dmetals.client=coc` flag is important since it configures Metals for
+usage with the `coc.nvim` client.
 
 ```scala mdoc:editor:vim
 
 ```
 
-## Learn more about vim-lsc
+### LSP commands key mapping
+
+`coc.nvim` doesn't come with a default key mapping for LSP commands, you need to configure it yourself.
+
+Here's a recommended configuration:
+
+```vim
+" ~/.vimrc
+" Configuration for coc.nvim
+
+" Smaller updatetime for CursorHold & CursorHoldI
+set updatetime=300
+
+" don't give |ins-completion-menu| messages.
+set shortmess+=c
+
+" always show signcolumns
+set signcolumn=yes
+
+" Some server have issues with backup files, see #649
+set nobackup
+set nowritebackup
+
+" Better display for messages
+set cmdheight=2
+
+" Use <c-space> for trigger completion.
+inoremap <silent><expr> <c-space> coc#refresh()
+
+" Use <cr> for confirm completion, `<C-g>u` means break undo chain at current position.
+" Coc only does snippet and additional edit on confirm.
+inoremap <expr> <cr> pumvisible() ? "\<C-y>" : "\<C-g>u\<CR>"
+
+" Use `[c` and `]c` for navigate diagnostics
+nmap <silent> [c <Plug>(coc-diagnostic-prev)
+nmap <silent> ]c <Plug>(coc-diagnostic-next)
+
+" Remap keys for gotos
+nmap <silent> gd <Plug>(coc-definition)
+nmap <silent> gy <Plug>(coc-type-definition)
+nmap <silent> gi <Plug>(coc-implementation)
+nmap <silent> gr <Plug>(coc-references)
+
+" Remap for do codeAction of current line
+nmap <leader>ac <Plug>(coc-codeaction)
+
+" Remap for do action format
+nnoremap <silent> F :call CocAction('format')<CR>
+
+" Use K for show documentation in preview window
+nnoremap <silent> K :call <SID>show_documentation()<CR>
+
+function! s:show_documentation()
+  if &filetype == 'vim'
+    execute 'h '.expand('<cword>')
+  else
+    call CocAction('doHover')
+  endif
+endfunction
+
+" Highlight symbol under cursor on CursorHold
+autocmd CursorHold * silent call CocActionAsync('highlight')
+
+" Remap for rename current word
+nmap <leader>rn <Plug>(coc-rename)
+
+" Show all diagnostics
+nnoremap <silent> <space>a  :<C-u>CocList diagnostics<cr>
+" Find symbol of current document
+nnoremap <silent> <space>o  :<C-u>CocList outline<cr>
+" Search workspace symbols
+nnoremap <silent> <space>s  :<C-u>CocList -I symbols<cr>
+" Do default action for next item.
+nnoremap <silent> <space>j  :<C-u>CocNext<CR>
+" Do default action for previous item.
+nnoremap <silent> <space>k  :<C-u>CocPrev<CR>
+" Resume latest coc list
+nnoremap <silent> <space>p  :<C-u>CocListResume<CR>
+```
+
+## Learn more about coc.nvim
 
-For comprehensive documentation about vim-lsc, run the following command.
+For comprehensive documentation about `coc.nvim`, run the following command.
 
 ```vim
-:help lsc
+:help coc-contents
 ```
 
 ## Customize goto definition
```

Comment:
I think this section can go, since it's extensively covered by the LSP commands key mapping one